docs/tutorials/developer-journey/level-1/1.3-first-dapp.md

```diff
@@ -0,0 +1,723 @@
+
+```
+import Text "mo:base/Text";
+```
+
+After all of these changes, your `main.mo` file should look like this:
+
+:::caution
+The following example is a **code snippet** that is part of a larger code file. This snippet may return an error if run on its own. To view the full code file that should be run, please see [final code](#final-code).
+:::
+
+```motoko
+import Text "mo:base/Text";
+
+actor {
+    var question: Text = "What is your favorite programming language?";
+
+}
+```
+
+Now, our code has an actor with one fixed variable and no methods. 
+
+For our next step, we'll add access methods. This is because in order to communicate with other canisters or any other external entity, actors must send and receive messages. 
+
+Specifically for our poll, we want to create the following functionality:
+
+- Obtain the current poll question.
+- Receive a list of possible options that can be voted on.
+- Submit a vote and have the dapp save the results. 
+- Reset the vote counts to re-run the poll. 
+
+### Defining the `getQuestion` method
+
+Let's start with the first functionality; obtain the current poll question. To do this, we'll create the `getQuestion` method using the following code:
+
+:::caution
+The following example is a **code snippet** that is part of a larger code file. This snippet may return an error if run on its own. To view the full code file that should be run, please see [final code](#final-code).
+:::
+
+```motoko
+public query func getQuestion() : async Text { 
+    question 
+};
+```
+
+:::info
+#### What does this code do?
+
+This `getQuestion` method takes the current value of the `question` variable and returns it to the caller. There are some important things regarding methods and their behavior to be aware of, such as:
+
+- All methods that return values on the IC have to be declared as `async` in order to allow asynchronous execution. 
+- Since the `getQuestion` method doesn't change any data, it can be declared as a `query` call. We'll dive further into this shortly. 
+- This method uses the `Text` type that we previously have already imported at the top of the file. 
+:::
+
+### Query calls vs. update calls
+
+A **query call** is executed on one node of a subnet. It is not possible to use query calls to alter data.
+
+An **update call** is executed on all nodes of a subnet. The result must pass through consensus on the subnet and has the ability to alter data. 
+
+A detailed comparison breakdown can be found below:
+
+|                      | Queries                    | Updates                |
+|----------------------|----------------------------|------------------------|
+| Resource consumption | Low                        | High                   |
+| Response times       | Fast (300ms-900ms)         | Slow (2s-10s)          |
+| Cost                 | Free                       | Cost cycles            |
```

Comment:
This table is nice. Since queries calls may incur a cost in the future, should we indicate the cost as "Low" instead of "Free"?

Reply:
Or "(Currently) free".

Do we also want to add a row on "Security"
|                                      | Queries                                        | Updates                                   |
|----------------------|--------------------------------|------------------------------|
| Security                      |  Results not certified by default | Results certified by default    |